hledger-web/Hledger/Web/Main.hs

```diff
@@ -79,11 +81,16 @@ web opts j = do
       let warpsettings = setHost (fromString h) (setPort p defaultSettings)
       case socket_ opts of
         Just s -> do
-          sock <- socket AF_UNIX Stream 0
-          setSocketOption sock ReuseAddr 1
-          bind sock $ SockAddrUnix s
-          listen sock maxListenQueue
-          Network.Wai.Handler.Warp.runSettingsSocket warpsettings sock app
+          bracket
+            (do
+                sock <- socket AF_UNIX Stream 0
+                setSocketOption sock ReuseAddr 1
+                bind sock $ SockAddrUnix s
+                listen sock maxListenQueue
+                return sock
+            )
+            (\_ -> removeFile s)
```

Comment:
Could use bracket_ here since the arg is ignored

Reply:
It is only ignored for the final operation, but still required for the in-between-computation.
`bracket` is `IO a -> (a -> IO b) -> (a -> IO c) -> IO c`, while something like `IO a -> IO b -> (a -> IO c) -> IO c` would be sufficient (so `bracket_` is not sufficient).